From 6622a83bc1c83bfcf861a0461ecd6ef6d769bf39 Mon Sep 17 00:00:00 2001
From: "Lain \"Fearyncess\" Yang" <fearyncess@aosc.io>
Date: Fri, 9 Jan 2026 13:10:51 +0800
Subject: [PATCH 23/69] BACKPORT: FROMLIST: loongarch: wire up memfd_secret
 system call

LoongArch supports ARCH_HAS_SET_DIRECT_MAP, therefore wire up the
memfd_secret system call, which depends on it.

Signed-off-by: Lain "Fearyncess" Yang <fearyncess@aosc.io>

Link: https://lore.kernel.org/loongarch/20260109051054.188030-1-fearyncess@aosc.io/

[Mingcong Bai: Resolved a minor merge conflict in
 arch/loongarch/kernel/Makefile.syscalls.]

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/loongarch/include/asm/unistd.h     | 1 +
 arch/loongarch/kernel/Makefile.syscalls | 3 +++
 tools/testing/selftests/mm/Makefile     | 2 +-
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/loongarch/include/asm/unistd.h b/arch/loongarch/include/asm/unistd.h
index e2c0f3d86c7bd5b149450ace1372f8cb6187154c..e7649c158248281b9938eb74cd59659567877861 100644
--- a/arch/loongarch/include/asm/unistd.h
+++ b/arch/loongarch/include/asm/unistd.h
@@ -10,5 +10,6 @@
 
 #define __ARCH_WANT_NEW_STAT
 #define __ARCH_WANT_SYS_CLONE
+#define __ARCH_WANT_MEMFD_SECRET
 
 #define NR_syscalls (__NR_syscalls)
diff --git a/arch/loongarch/kernel/Makefile.syscalls b/arch/loongarch/kernel/Makefile.syscalls
index cd46c2b69c7fde9cda7d0d11174d70570ef00a52..43abb068a5c327cc9a5a4e56c95952a3a519da94 100644
--- a/arch/loongarch/kernel/Makefile.syscalls
+++ b/arch/loongarch/kernel/Makefile.syscalls
@@ -3,3 +3,6 @@
 # No special ABIs on loongarch so far
 syscall_abis_32 +=
 syscall_abis_64 +=
+# Add memfd_secret explictly for la64 and la32
+syscall_abis_32 += memfd_secret
+syscall_abis_64 += memfd_secret
diff --git a/tools/testing/selftests/mm/Makefile b/tools/testing/selftests/mm/Makefile
index eaf9312097f7babcf48131b9dff66b3d68bc3781..79582438efc4d64584ddd8a429d79ced9cf8b5c9 100644
--- a/tools/testing/selftests/mm/Makefile
+++ b/tools/testing/selftests/mm/Makefile
@@ -72,7 +72,7 @@ TEST_GEN_FILES += madv_populate
 TEST_GEN_FILES += map_fixed_noreplace
 TEST_GEN_FILES += map_hugetlb
 TEST_GEN_FILES += map_populate
-ifneq (,$(filter $(ARCH),arm64 riscv riscv64 x86 x86_64))
+ifneq (,$(filter $(ARCH),arm64 riscv riscv64 x86 x86_64 loongarch32 loongarch64))
 TEST_GEN_FILES += memfd_secret
 endif
 TEST_GEN_FILES += migration
-- 
2.53.0

